jobserver/releases.py

```diff
@@ -125,38 +164,60 @@ def handle_file_upload(release, backend, user, upload, filename, **kwargs):
     # TODO: enforce a max upload size?
     data = upload.read()
     calculated_hash = hashlib.sha256(data).hexdigest()
-    check_not_already_uploaded(filename, calculated_hash, backend)
+    relative_path, absolute_path = _build_paths(release, filename, data)
 
-    # We group the directory structure by workspace for 2 reasons:
-    # 1. avoid dumping everything in one big directory.
-    # 2. much easier for human operators to navigate on disk if needed.
-    relative_path = (
-        Path(release.workspace.name) / "releases" / str(release.id) / filename
-    )
-    absolute_path = absolute_file_path(relative_path)
-    absolute_path.parent.mkdir(parents=True, exist_ok=True)
-    absolute_path.write_bytes(data)
-
-    mtime = datetime.fromtimestamp(absolute_path.stat().st_mtime, tz=timezone.utc)
-
-    try:
-        rfile = ReleaseFile.objects.create(
-            release=release,
-            workspace=release.workspace,
-            created_by=user,
-            name=filename,
-            path=str(relative_path),
-            filehash=calculated_hash,
-            mtime=mtime,
-            size=absolute_path.stat().st_size,
-            **kwargs,
+    # Check if this filename for this release has been uploaded before.
+    existing = ReleaseFile.objects.filter(
```

Comment:
`existing` is the wrong name here I think. It only makes sense for the following if clause, it reads a bit oddly for the rest of the code.

I'd prefer `rfile` or similar?